flocker/node/agents/blockdevice.py

```diff
@@ -431,6 +496,13 @@ class DestroyBlockDeviceDataset(PRecord):
     dataset_id = field(type=UUID, mandatory=True)
     blockdevice_id = field(type=unicode, mandatory=True)
 
+    @classmethod
+    def from_dataset(cls, discovered_dataset, desired_dataset):
```

Comment:
Maybe these should be called... `from_discovered_and_desired`? `from_state_and_config`?